compiler/src/dmd/root/string.d

```diff
@@ -238,6 +238,30 @@ int dstrcmp()( scope const char[] s1, scope const char[] s2 ) @trusted
     return s1.length < s2.length ? -1 : (s1.length > s2.length);
 }
 
+
+/**
+ * Checks if a given path points to file with an extension.
+ */
+bool hasExtension(char[] path)
```

Comment:
You could just use `dmd.root.filename : FileName.ext`: `FileName.ext(path).length`.